config/opal_config_pthreads.m4

```diff
@@ -251,9 +252,10 @@ AC_PROVIDE_IFELSE([AC_PROG_CC],
                   [OPAL_INTL_POSIX_THREADS_PLAIN_C],
                   [opal_pthread_c_success=1])
 
-AC_PROVIDE_IFELSE([AC_PROG_CXX],
-                  [OPAL_INTL_POSIX_THREADS_PLAIN_CXX],
-                  [opal_pthread_cxx_success=1])
+AS_IF([test "$WANT_MPI_CXX_SUPPORT" = 1],
```

Comment:
I think `$CXX` will be set to `no` (or empty?) if there is no valid C++ compiler.

/me goes to test...

It looks like configure sets `$CXX` to `g++` if it can't find a valid C++ compiler (!).  But then if you try to run it, for example, it will fail (e.g., if it can't be found). 😞 

The only place we have C++ in the code base is in the MPI C++ bindings (and examples), but those aren't even built by default any more.

I wonder if this whole C++ section of configure is in need of a bit of revamping / simplification...